src/cmd/linuxkit/pkglib/pkglib.go

```diff
@@ -169,6 +172,43 @@ func NewFromCLI(fs *flag.FlagSet, args ...string) (Pkg, error) {
 		}
 	})
 
+	var srcHashes string
+	var sources []string
+	if len(pi.Sources) != 0 {
+		// If we have external sources ensure the package path is included first
+		sources = []string{pkgPath + ":/"}
+	}
```

Comment:
If we made it so that sources always contains at least the pkgPath (and nothing else if `pi.Sources` is unused) then wouldn't that remove a bunch of conditionality elsewhere -- e.g. the building bit would then always pass the context on stdin even in the single source case.

Reply:
Almost certainly. I contemplated it but initially wanted to add this feature but leave the other code path untouched, but maybe now it's better to always use the build context

Reply:
I think it is necessary to leave the behaviour the same, but changing the internals as needed to make the new stuff fit cleanly is the right thing to do.